openapi.yaml

```diff
@@ -16,175 +16,511 @@ host: api.oregonstate.edu
+  players:
+    type: array
+    items:
+      $ref: '#/definitions/player'
+  player:
+    type: object
+    properties:
+      cards:
+        $ref: '#/definitions/cards'
+      bet:
+        $ref: '#/definitions/playerBet'
+      status:
+        $ref: '#/definitions/playerStatus'
+  playerBet:
+    type: integer
+    minimum: 0
+    description: Each player's bet
+  playerStatus:
+    type: string
+    enum:
+      - "called"
+      - "raised"
+      - "checked"
+      - "folded"
+  playerResource:
+    properties:
+      id:
+        $ref: '#/definitions/playerID'
+      type:
+        $ref: '#/definitions/playerType'
       attributes:
+        type: object
         properties:
-          name:
-            type: string
-            description: Name of pet
-            example: Hedwig
-          owner:
-            type: string
-            description: Name of owner
-            example: Harry Potter
-          species:
-            type: string
-            description: Species of pet
-            example: Owl
-  PetResult:
+          players:
+            $ref: '#/definitions/players'
+  card:
+    properties:
+      attributes:
+        type: object
+        properties:
+          cardNumber:
+            $ref: '#/definitions/cardNumber'
+          cardSuit:
+            $ref: '#/definitions/cardSuit'
+  cardNumber:
+    type: string
+    enum: 
+      - 'A'
+      - '2'
+      - '3'
+      - '4'
+      - '5'
+      - '6'
+      - '7'
+      - '8'
+      - '9'
+      - '10'
+      - 'J'
+      - 'Q'
+      - 'K'
+  cardSuit:
+    type: string
+    enum:
+      - 'Diamond'
+      - 'Club'
+      - 'Spade'
+      - 'Heart'
+      
+  GamePostBody:
+    properties:
+      data:
+        type: object
+        properties:
+          type:
+            $ref: '#/definitions/gameType'
+          attributes:
+            type: object
+            properties:
+              playerNum:
+                $ref: '#/definitions/playerNum'
+              minimumBet:
+                $ref: '#/definitions/minimumBet'
+              maximumBet:
+                $ref: '#/definitions/maximumBet'
+            required:
+              - playerNum
+              - minimumBet
+              - maximumBet
+            additionalProperties: false
+        required: 
+          - type
+          - attributes
+        additionalProperties: false
+    required:
+      - data
+    additionalProperties: false
+  
+  GameUpdateBody:
+    properties:
+      data:
+        type: object
+        properties:
+          type:
+            $ref: '#/definitions/gameType'
+          attributes:
+            type: object
+            properties:
+              round:
+                $ref: '#/definitions/round'
+              players:
+                $ref: '#/definitions/players'
+              tableCards:
+                $ref: '#/definitions/cards'
+            required:
+              - round
+              - players
+              - tableCards
```

Comment:
When will the game update be called? Before the game start or during the game or both? Here are some scenarios I can think of:

1. After the game created and before the game starts: players are possible to join to and leave from a game, so we can call this function to update the players list. Is it required to contain `round` or `tableCards` in the data body at this stage? Also, should we allow users to modify min/max bet and players number at this stage?
2. During the game, the players list shouldn't be modified which means the `players` field shouldn't be required here. However, we can still update the game round and table cards by using this method.